src/ocm/components/hosts/HostsDiscoveryTable.tsx

```diff
@@ -74,6 +85,20 @@ const HostsDiscoveryTable: React.FC<HostsDiscoveryTableProps> = ({
     [onEditHost, actionChecks.canEditHostname, actionChecks.canEditRole, onEditRole, cluster],
   );
 
+  React.useEffect(() => {
+    const forceRole = async () => {
+      cluster.hosts?.forEach(async (host) => {
+        if (host.role !== 'master') {
+          const { data } = await HostsService.updateRole(cluster.id, host.id, 'master');
+          dispatch(updateHost(data));
+        }
```

Comment:
Would be better written and allow for better error handling if it would be written with [`for await...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of)

```suggestion
      for await (const host of cluster.hosts) {
        if (host.role !== 'master') {
          try {
            const { data } = await HostsService.updateRole(cluster.id, host.id, 'master');
            dispatch(updateHost(data));
          } catch (error) {
            // here comes the error handling part...
          }
        }
```